addons/point_of_sale/static/src/js/Screens/ClientListScreen/ClientDetailsEdit.js

```diff
@@ -44,7 +44,7 @@ odoo.define('point_of_sale.ClientDetailsEdit', function(require) {
                     processedChanges[key] = value;
                 }
             }
-            if (!processedChanges.name) {
+            if (!this.props.partner.name && !processedChanges.name) {
```

Comment:
@fedegobea You are right. But I think this check is more appropriate: `processedChanges.name === ''`. Can you verify?

The goal of the clause is that if the user made a change in the name, it should not be allowed to be an empty string. However, the use of `!processedChanges.name` was a mistake because it is also `true` when `.name` is `undefined` -- meaning, name is unchanged. So the fix will be to check if the 'proposed name change' is an empty string, if so, show an error message.